### PR TITLE
Add support for DD kwarg in qss

### PR DESCRIPTION
**Pull request by paaige**

### CODE REVIEW

**Review by knsmith — APPROVED** [review]

File: qiskit-superstaq/qiskit_superstaq/superstaq_backend.py [hunk 1]
```diff
@@ -144,12 +144,14 @@ def retrieve_job(self, job_id: str) -> qss.SuperstaqJob:
     def compile(
         self,
         circuits: Union[qiskit.QuantumCircuit, Sequence[qiskit.QuantumCircuit]],
+        dynamical_decoupling: bool = False,
```
Comment (singular-value) [inline qiskit-superstaq/qiskit_superstaq/superstaq_backend.py:147]: do we want it to default to True?

File: qiskit-superstaq/qiskit_superstaq/superstaq_backend.py [hunk 2]
```diff
@@ -179,6 +181,7 @@ def compile(
     def _get_compile_request_json(
         self,
         circuits: Union[qiskit.QuantumCircuit, Sequence[qiskit.QuantumCircuit]],
+        dynamical_decoupling: bool = False,
```
Comment (richrines1) [inline qiskit-superstaq/qiskit_superstaq/superstaq_backend.py:184]: this shouldn't need to be added here, it'd be better to put this in the options dict

**Review by knsmith — APPROVED** [review]

**Review by richrines1 — APPROVED** [review]
lgtm! thanks for the quick fix :) we should probably think of a good way to test this but for now let's just get it in

also definitely not for this pr but we should probably add dynamical_decoupling as an argument to qss.superstaq_provider.ibmq_compile and css.service.ibmq_compile as well

File: qiskit-superstaq/qiskit_superstaq/superstaq_backend.py [hunk 3]
```diff
@@ -150,6 +150,7 @@ def compile(
 
         Args:
             circuits: The qiskit QuantumCircuit(s) to compile.
+            dynamical_decoupling: Setting to add dynamical decoupling to the circuit.
```
Comment (richrines1) [inline qiskit-superstaq/qiskit_superstaq/superstaq_backend.py:153]:
```suggestion
```

**Pull request merged**